tests/components/shelly/test_switch.py

```diff
@@ -56,6 +65,113 @@ async def test_block_device_services(
     assert hass.states.get("switch.test_name_channel_1").state == STATE_OFF
 
 
+@pytest.mark.parametrize("model", MOTION_MODELS)
+async def test_block_motion_switch(
+    hass: HomeAssistant,
+    model: str,
+    mock_block_device: Mock,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Test Shelly motion active turn on/off services."""
+    entity_id = "switch.test_name_motion_active"
+    await init_integration(hass, 1, sleep_period=1000, model=model)
+
+    # Make device online
+    mock_block_device.mock_online()
+    await hass.async_block_till_done(wait_background_tasks=True)
+
+    assert get_entity_state(hass, entity_id) == STATE_ON
+
+    # turn off
+    await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: entity_id},
+        blocking=True,
+    )
+    monkeypatch.setattr(mock_block_device.blocks[MOTION_BLOCK_ID], "motionActive", 0)
+    mock_block_device.mock_update()
+
+    mock_block_device.set_shelly_motion_detection.assert_called_once_with(False)
+    assert get_entity_state(hass, entity_id) == STATE_OFF
+
+    # turn on
+    mock_block_device.set_shelly_motion_detection.reset_mock()
+    await hass.services.async_call(
+        SWITCH_DOMAIN,
+        SERVICE_TURN_ON,
+        {ATTR_ENTITY_ID: entity_id},
+        blocking=True,
+    )
+    monkeypatch.setattr(mock_block_device.blocks[MOTION_BLOCK_ID], "motionActive", 1)
+    mock_block_device.mock_update()
+
+    mock_block_device.set_shelly_motion_detection.assert_called_once_with(True)
+    assert get_entity_state(hass, entity_id) == STATE_ON
+
+
+@pytest.mark.parametrize("model", MOTION_MODELS)
+async def test_block_restored_motion_switch(
+    hass: HomeAssistant,
+    model: str,
+    mock_block_device: Mock,
+    device_reg: DeviceRegistry,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Test block restored motion active switch."""
+    entry = await init_integration(
+        hass, 1, sleep_period=1000, model=model, skip_setup=True
+    )
+    register_device(device_reg, entry)
+    entity_id = register_entity(
+        hass, SWITCH_DOMAIN, "test_name_motion_active", "sensor_0-motionActive", entry
+    )
+
+    mock_restore_cache(hass, [State(entity_id, STATE_OFF)])
+    monkeypatch.setattr(mock_block_device, "initialized", False)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert get_entity_state(hass, entity_id) == STATE_OFF
+
+    # Make device online
+    monkeypatch.setattr(mock_block_device, "initialized", True)
+    mock_block_device.mock_online(wait_background_tasks=True)
```

Comment:
Sorry, I accidentally commented on the wrong line and, the `wait_background_tasks` should be a parameter for the `async_block_till_done`